src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/SystemService.java

```diff
@@ -163,6 +179,55 @@ private String getExceptionMessage(String name, String url, String message) {
         return csrBytes;
     }
 
+    /**
+     * Get configuration anchor file
+     * @return
+     * @throws AnchorNotFoundException if anchor file is not found
+     */
+    public AnchorFile getAnchorFile() throws AnchorNotFoundException {
+        AnchorFile anchorFile = new AnchorFile(calculateAnchorHexHash(readAnchorFile()));
+        ConfigurationAnchorV2 anchor = anchorRepository.loadAnchorFromFile();
+        anchorFile.setCreatedAt(FormatUtils.fromDateToOffsetDateTime(anchor.getGeneratedAt()));
+        return anchorFile;
```

Comment:
Adding anchor filename into `AnchorFile` dto does not help much since the filename is never used when the dto is returned. The filename is used only when the anchor file is downloaded and in that case the service layer returns the file as `byte[]`. You're right that reading the file and getting the filename require two filesystem operations. However, downloading the anchor file is a relatively rare use case so the additional file system hit should not affect the performance.

Reply:
You are right it isn't used. Got confused about the reading and loading anchor from a file. Maybe the dto could be used in both cases by adding the bytes to it too. 

But yeah the anchor download is a rare use case so this is good 👍